repos/system_upgrade/el7toel8/actors/detectgrubconfigerror/libraries/scanner.py

```diff
@@ -0,0 +1,14 @@
+import re
+
+
+def detect_config_error(conf_file):
+    '''
+    Checks grub configuration for syntax error in GRUB_CMDLINE_LINUX value.
+
+    :return: Function returns True if error was detected, otherwise False.
+    '''
+    with open(conf_file, 'r') as f:
+        config = f.read()
+
+    pattern = 'GRUB_CMDLINE_LINUX="[^"]+"\s(?=[^A-Z]+)'
```

Comment:
```suggestion
    pattern = r'GRUB_CMDLINE_LINUX="[^"]+"(?!(\s*$)|(\s+GRUB))'
```
Your regex incorrectly marked the following as **having** syntax error:
```
GRUB_TERMINAL_OUTPUT="console"
GRUB_CMDLINE_LINUX="console=tty0 crashkernel=auto console=ttyS0,115200n8 no_timer_check net.ifnames=0" 
GRUB_DISABLE_RECOVERY="true"
```
Or, the regex incorrectly marked the following as **not** having syntax error:
```
GRUB_TERMINAL_OUTPUT="console" GRUB_CMDLINE_LINUX="vconsole.font=latarcyrheb-sun16 vconsole.keymap=us "rhgb quiet GRUB_DISABLE_RECOVERY="true"
```
Please add these two files to the tests.